doc/cats.adoc

```diff
@@ -675,18 +675,67 @@ TODO
 
 TODO
 
+=== Lazy sequences
+
+TODO
 
 === Vector
 
 TODO
 
+=== Set
+
+TODO
+
+=== Map
+
+TODO
+
+=== Validation
+
+The validation type is similar to Either or Exception types but it doesn't implement a Monad instance. It
+has two constructors: `ok` and `fail`, representing success and failure respectively.
+
+[source, clojure]
+----
+(require '[cats.applicative.validation :as v])
+(require '[cats.core :as m])
+
+(v/ok 42)
+;;=> #<Ok 42>
+
+(v/fail [])
+;;=> #<Fail []>
+----
+
+It implements the Applicative protocol, and its intended usage is as an Applicative. Applying Validation
+values together errs on the side of the failure, and applying failures together aggregates their values
+using the Semigroup's `mappend` function.
+
+[source, clojure]
+----
+(require '[cats.applicative.validation :as v])
+(require '[cats.core :as m])
+
+(m/fapply (v/ok 42) (v/fail "OH NOES"))
```

Comment:
It does make sense. Note that in the Validation applicative function application is only run with two Ok instances. It [follows the applicative laws](https://github.com/funcool/cats/blob/master/test/cats/applicative/validation_spec.cljc), see [this Haskell library](https://github.com/NICTA/validation) for different flavours of validation types.